types/ember-data/index.d.ts

```diff
@@ -919,10 +919,10 @@ export namespace DS {
      * it easy to create data bindings with the `PromiseArray` that will be
      * updated when the promise resolves.
      */
-    interface PromiseArray<T>
+    interface PromiseArray<T, ArrayType = Ember.Array<T>>
```

Comment:
As it stands, looks like the `ArrayType` here *could* be passed *any* type, even though it defaults to `Ember.Array<T>`. Should it be `extends Array<T>` or similar?